### PR TITLE
Structured binding support for IKeyValuePair

### PR DESCRIPTION
**Pull request by oldnewthing**
Addresses issue #586

### CODE REVIEW

File: strings/base_collections_map.h [hunk 1]
```diff
@@ -66,3 +66,36 @@ WINRT_EXPORT namespace winrt
         return make<impl::observable_map<K, V, std::unordered_map<K, V, Hash, KeyEqual, Allocator>>>(std::move(values));
     }
 }
+
```
Comment (oldnewthing) [inline strings/base_collections_map.h:69]: I wasn't sure whether to put it here or in `base_collections.h`. Feel free to edit.

File: test/old_tests/UnitTests/range_for.cpp [hunk 2]
```diff
@@ -69,6 +69,21 @@ TEST_CASE("range_for,IIterable,IKeyValuePair")
...
+        REQUIRE(key == i.Key());
+        REQUIRE(value == i.Value());
+    }
+
+    REQUIRE(result == values);
+
+    // Do it again with structured bindings.
+    result.clear();
+
+    for (auto&& [key, value] : c)
```
Comment (DefaultRyan) [inline test/old_tests/UnitTests/range_for.cpp:84]: Nice. Been awhile and I'd forgotten about this syntax. :)

File: strings/base_collections_map.h (hunk 1 above)
Comment (kennykerr) [reply strings/base_collections_map.h:69]: Works for me. 👍

**Pull request merged**